packages/mosaic-tanstack-table-core/src/data-table.ts

```diff
@@ -506,49 +549,75 @@ export class MosaicDataTable<
 type FacetClientConfig<TResult extends Array<any>> = {
   filterBy?: Selection;
   coordinator?: Coordinator | null;
-  table: string;
+  source: MosaicTableSource;
   column: string;
   getFilterExpressions?: () => Array<mSql.FilterExpr>;
   onResult: (...values: TResult) => void;
 };
 
-/**
- * This is a helper Mosaic Client to query unique values for a given column.
- */
 export class UniqueColumnValuesClient extends MosaicClient {
-  from: string;
+  source: MosaicTableSource;
   column: string;
   getFilterExpressions?: () => Array<mSql.FilterExpr>;
   onResult: (values: Array<unknown>) => void;
+  lastQuerySql?: string; // Cache for deduping
 
   constructor(options: FacetClientConfig<Array<unknown>>) {
     super(options.filterBy);
-
-    if (options.coordinator) {
-      this.coordinator = options.coordinator;
-    } else {
-      this.coordinator = defaultCoordinator();
-    }
-
-    this.from = options.table;
+    this.coordinator = options.coordinator ?? defaultCoordinator();
+    this.source = options.source;
     this.column = options.column;
     this.getFilterExpressions = options.getFilterExpressions;
     this.onResult = options.onResult;
   }
 
   connect(): void {
-    this.coordinator?.connect(this);
+    if (!this.coordinator) {
+      this.coordinator = defaultCoordinator();
```

Comment:
Is this needed? By this point, higher up in the tree (at `updateOptions()`), the coordinator should have already been set.